src/main/java/software/amazon/nio/spi/s3/S3ClientProvider.java

```diff
@@ -126,9 +126,9 @@ S3CrtAsyncClientBuilder configureCrtClient() {
             asyncClientBuilder.endpointOverride(endpointUri);
         }
 
-        var credentials = configuration.getCredentials();
```

Comment:
Deleting this is a potential breaking change. Your later javadoc suggests that both could be provided and credentialsProvider has precedent. If a credentials provider is not supplied then I think people may still want to be able to provide credentials.